packages/wattles/lib/src/drivers/memory_driver.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:wattles/wattles.dart';
+
+/// {@template sql_driver}
+/// Driver for storing data in memory.
+/// {@endtemplate}
+class MemoryDriver extends DatabaseDriver {
+  final Map<String, List<Map<String, dynamic>>> _data = {};
+
+  @override
+  Future<int> insert(
+    String table,
+    Schema rootSchema,
+    SchemaInstance instance,
+  ) async {
+    final collection = _data.putIfAbsent(table, () => []);
+    final primaryKey = rootSchema.properties.firstWhere(isPrimary);
+
+    final fields = {
+      for (final prop in rootSchema.properties.where(isNotPrimary))
+        prop.fromKey: instance.get(prop)!,
+    };
+
+    final map = fields.map((key, value) => MapEntry(key, value.value));
+    collection.add(map);
+
+    final id = collection.length;
+
+    instance.set(primaryKey, id);
+    return map[primaryKey.fromKey] = id;
+  }
+
+  @override
+  Future<int> update(
+    String table,
+    Schema rootSchema,
+    SchemaInstance instance, {
+    required Query query,
+  }) async {
+    final collection = _data.putIfAbsent(table, () => []);
+
+    final records = collection.where((e) {
+      return query.wheres.any(
```

Comment:
The `query.wheres` could maybe be moved into a private method to reduce redundancy?

Something like:
```suggestion
   List<Map<String, dynamic>> _queryCollection(collection, query) {
      return query.wheres.any(
         (wheres) => wheres.every((where) => where.check(e)),
      );
   }
```

Reply:
I just noticed there are more things wrong, so need to refactor it a bit more! Nice catch!